### ISSUE TITLE
Request: jekyll-assets

### ISSUE THREAD

**Issue by dynobo** [FauDroids/MrHyde#68]
Hi,

is it possible for you to install additional jekyll-plugins on the Build Server?

I currently can't build, because the server is missing [jekyll-assets](https://jekyll.github.io/jekyll-assets/).

Thanks!

**Comment by s1hofmann** [comment]
Should be done. Could you give it a try and ping back if it works?

**Comment by dynobo** [comment]
Not quite, yet. I don't get the log error as before, instead it shows the error below.
Do I have to delete cache somehow or similar things?

![croppedscreenshot_1488189964716_](https://cloud.githubusercontent.com/assets/11071876/23357157/e21802c8-fcdc-11e6-9288-a144f26c486a.png)

**Comment by dynobo** [comment]
Closed by accident

**Comment by s1hofmann** [comment]
Sorry for the delay, but here's a little follow up to this issue.

Additional Jekyll dependencies were resolved quite fast, but as your screenshot shows, the following problem is caused by version control on the backend.

    > error: cannot apply binary patch to 'font/open-sans-v13-latin-regular.woff2' without full index line
    > error: font/open-sans-v13-latin-regular.woff2: patch does not apply'

Git fails to apply binary patches, which is why it discards the whole patch. To resolve this issue we'll need some more investigation, but we're trying to sort things out.

**Comment by dynobo** [comment]
Thanks for this Info!
Is there anything I could do as a work around till then? Like excluding these files from being patched somehow?

**Comment by Maddoc42** [comment]
Hm, why is there a patch being applied to the font files in the first place? Have you modified those files on your device?

**Comment by dynobo** [comment]
To be honest, I have no idea. I just downloaded the files from google fonts. Maybe I did update them later. But I didn't apply some kind of patch, at least not on purpose.

**Comment by Maddoc42** [comment]
Could you check your git diff, are the font files listed there?  You can see the diff in the commit screen.

**Comment by Maddoc42** [comment]
@dynobo to clarify the question above: did you download the font files on your mobile device or have you added them to the repo in a previous commit?

Font files typically don't change that often. Maybe the files got corrupted by accident ... if you do a clean checkout with MrHyde, does the preview work?

So many questions :smile: :see_no_evil:

**Comment by dynobo** [comment]
@Maddoc42 To check, I made forced a reset in my remote repo, and also delete local repo on android + reinstalled MrHyde. Sadly, now I'm facing another problem #72 . I'll report back about the font-problem, as soon as I'm able to checkout my repo again in MrHyde ;) Thanks for your support!

**Comment by dynobo** [comment]
Okay, I did a complete reset of the remote repo like this:
 
```
rm -rf .git
git init
git add .
git commit -m "Initial commit after reset"
git remote add origin git://... 
git push --force --set-upstream origin master
```

Now only a this initial commit is visible in gitlog. 

Also I completely re-installed MrHyde on Android, including deleting local repo.

Still I got the same VCS error.

**Comment by Maddoc42** [comment]
Did you make any changes in MrHyde (or any other app ...)? When you try to commit changes, are there any diffs listed?

I still don't get why there should be a patch applied at all. @s1hofmann the error message is from git trying to apply a patch, correct?

@dynobo as a last resort, do you have a public version of your repo we can test? Or even better a minimal version which shows this error?

**Comment by s1hofmann** [comment]
Okay, so here's another insight:

I just had a look at the generated patch file.
According to this, the patch file is generated because file permissions changed.

Here's the generated patch file:
```
 88 diff --git a/_assets/fonts/open-sans-v13-latin-600.woff2 b/_assets/fonts/open-sans-v13-latin-600.woff2
 89 old mode 100755
 90 new mode 100644
 91 Binary files differ
 92 diff --git a/_assets/fonts/open-sans-v13-latin-regular.woff b/_assets/fonts/open-sans-v13-latin-regular.woff
 93 old mode 100755
 94 new mode 100644
 95 Binary files differ
 96 diff --git a/_assets/fonts/open-sans-v13-latin-regular.woff2 b/_assets/fonts/open-sans-v13-latin-regular.woff2
 97 old mode 100755
 98 new mode 100644
 99 Binary files differ
```

Consulting StackOverlow has lead me to this post [here](http://stackoverflow.com/questions/1257592/how-do-i-remove-files-saying-old-mode-100755-new-mode-100644-from-unstaged-cha), which pretty much sums up our problem here.

@Maddoc42 Want to discuss things this evening?

**Comment by s1hofmann** [comment]
@dynobo We had a chat about this issue yesterday and were both wondering about the file permission thing.
Could you look up your local file permissions and how git checks in your files?
Which service are you using? Gitlab, GitHub or BitBucket?

If your local file permissions are set to 755, try changing them to 644 as a first simple fix to avoid MrHyde from generating a patch file just because file permissions changed.
Adding files to your .gitignore won't help, because I'm assuming your whole repository would be affected.

To further investigate on this issue, it would be super awesome if we could access a public repository of yours for some additional tests. Any chance you could provide one?

**Comment by dynobo** [comment]
@s1hofmann My repo is on gitlab. I would prefere to not make it public, but can you sent me a pm with your usernames/mail-addresses there? Then I would add you as members. 

Most of my files are 755, including the font files. I chmod fonts to 644, again reset the repo, still same error. But I don't no how to check, if permissions are preserved in remote repo after push.

**Comment by s1hofmann** [comment]
I just checked the logs again, the file permission problem is still there.
I'll do a separate check using GitLab, if this problem is reproducible, I'll try to fix it on the backend.
